homeassistant/components/switchbee/climate.py

```diff
@@ -0,0 +1,189 @@
+
+    def __init__(
+        self,
+        device: SwitchBeeThermostat,
+        coordinator: SwitchBeeCoordinator,
+    ) -> None:
+        """Initialize the Switchbee switch."""
+        super().__init__(device, coordinator)
+        # set HVAC capabilities
+        self._attr_max_temp = device.max_temperature
+        self._attr_min_temp = device.min_temperature
+        self._attr_temperature_unit = HVAC_UNIT_SB_TO_HASS[device.unit]
+        self._attr_hvac_modes = [HVAC_MODE_SB_TO_HASS[mode] for mode in device.modes]
+        self._attr_hvac_modes.append(HVACMode.OFF)
+        self._update_attrs_from_coordinator()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_attrs_from_coordinator()
+        super()._handle_coordinator_update()
+
+    def _update_attrs_from_coordinator(self) -> None:
+
+        coordinator_device = self._get_coordinator_device()
+
+        self._attr_hvac_mode: HVACMode = (
+            HVACMode.OFF
+            if coordinator_device.state == ApiStateCommand.OFF
+            else HVAC_MODE_SB_TO_HASS[coordinator_device.mode]
+        )
+        self._attr_fan_mode = FAN_SB_TO_HASS[coordinator_device.fan]
+        self._attr_current_temperature = coordinator_device.temperature
+        self._attr_target_temperature = coordinator_device.target_temperature
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set hvac mode."""
+
+        if hvac_mode == HVACMode.OFF:
+            await self._operate(power=ApiStateCommand.OFF)
+        else:
+            await self._operate(
+                power=ApiStateCommand.ON, mode=HVAC_MODE_HASS_TO_SB[hvac_mode]
+            )
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        await self._operate(target_temperature=kwargs[ATTR_TEMPERATURE])
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        """Set AC fan mode."""
+        await self._operate(fan=FAN_HASS_TO_SB[fan_mode])
+
+    async def _operate(
+        self,
+        power: str | None = None,
+        mode: str | None = None,
+        fan: str | None = None,
+        target_temperature: int | None = None,
+    ) -> None:
+        """Send request to central unit."""
+
+        assert isinstance(self.hvac_mode, HVACMode)
+        assert isinstance(self.fan_mode, str)
+        assert isinstance(self.target_temperature, int)
+
+        new_power = power
+        if not new_power:
+            new_power = (
+                ApiStateCommand.OFF
+                if self.hvac_mode == HVACMode.OFF
+                else ApiStateCommand.ON
+            )
+
+        new_mode = mode if mode else HVAC_MODE_HASS_TO_SB[self.hvac_mode]
+        new_fan = fan if fan else FAN_HASS_TO_SB[self.fan_mode]
+        new_temperature = (
+            target_temperature if target_temperature else self.target_temperature
+        )
```

Comment:
If you adjust it here, then you can drop the three asserts above:
```suggestion
        new_temperature = (
            int(self.target_temperature) if target_temperature is None else target_temperature
        )
```

Reply:
I remember it is not recommended to use ternary operator over multiple lines, so this can also be modified to:
```python
new_temperature = self.target_temperature
if target_temperature is not None:
    new_temperature = target_temperature
```

Reply:
Yes I forgot about that. I was focusing on the type hints too much.

I think in this case what would be better is to avoid the four new variables:

```python
        if power is None:
            power = ApiStateCommand.ON
            if self.hvac_mode == HVACMode.OFF:
                power = ApiStateCommand.OFF
        if mode is None:
            mode = HVAC_MODE_HASS_TO_SB[self.hvac_mode]
        if fan is None:
            fan = FAN_HASS_TO_SB[self.fan_mode]
        if target_temperature is None:
            target_temperature = int(self.target_temperature or 0)

        state: dict[str, int | str] = {
            ApiAttribute.POWER: power,
            ApiAttribute.MODE: mode,
            ApiAttribute.FAN: fan,
            ApiAttribute.CONFIGURED_TEMPERATURE: target_temperature,
        }
```

Reply:
I agree, you suggestion is also more clear to understand 👍